pkg/metrics/docs/generator.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+const MARKDOWN_OUTPUT = "doc/metrics_catalog.md"
+
+type Metric struct {
+	Name        string `yaml:"name"`
+	Type        string `yaml:"type"`
+	Description string `yaml:"description,omitempty"`
+	File        string `yaml:"file,omitempty"`
+}
+
+type MetricType struct {
+	enabled bool
+	name    string
+}
+
+var metricTypes = map[string]MetricType{
```

Comment:
https://prometheus.io/docs/concepts/metric_types/

Reply:
Maybe worth including this as a comment so it doesn't get lost